File: su-lim/13305.py
import sys

input = sys.stdin.readline

N = int(input())
distance = list(map(int, input().split())) # 도시 사이 거리
oil_price = list(map(int, input().split())) # 도시 마다 기름 값

result = 0;
current = oil_price[0];

for i in range(N-1):

	if current > oil_price[i]:
		current = oil_price[i]

	result += current * distance[i]

print(result)

File: YEON/BAEKJOON/13305.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split("\n");

const N = input[0];
const km = input[1].split(" ").map( i => BigInt(i));
const cost = input[2].split(" ").map( i => BigInt(i));

let i = 0;
let current = cost[i];
let total = 0n;

while(true) {
    if(current >  cost[i]) {
        current =  cost[i];
    }
    total += current * km[i];
    i+=1;

    if(i === km.length) {
        break;
    }
}

console.log(String(total));
